src/main/scala/com/scality/clueso/compact/TableFilesCompactor.scala

```diff
@@ -0,0 +1,163 @@
+      }
+    }
+
+    subpartitionValues
+  }
+
+  def removeSubpartitionsFromLanding(landingPartitionPath: String, subPartToCompact: Array[Long]): Unit = {
+    val subpartitionPaths = fs.listStatus(new Path(landingPartitionPath), new PathFilter {
+      override def accept(path: Path): Boolean = {
+        if (fs.isDirectory(path) && path.getName.matches(partDirnamePattern.pattern.pattern())) {
+          val regexMatch = partDirnamePattern.findFirstMatchIn(path.getName)
+          regexMatch.foldLeft(false) { (v, rm) => v || subPartToCompact.contains(rm.group(2).toLong) }
+        } else {
+          false
+        }
+      }
+    })
+
+    subpartitionPaths.foreach(fileStatus => fs.delete(fileStatus.getPath, true))
+  }
+
+  /**
+    * Merges the contents of /landing/<partitionColumnName>=<partitionValue>/`*`/`*`.parquet and deletes
+    * /landing/_spark_metadata. /landing/bucket=mybucketname/ will be composed of multiple subpartitions
+    * with format maxOpIndex=<N>. If only of such subpartitions is existent, the partition bucket=.. won't be
+    * merged into /staging. This can be forced by setting <force> to `true`
+    *
+    * @param partitionColumnName usually equal to 'bucket'
+    * @param partitionValue      usually equal to the name of the bucket
+    * @param numPartitions       num of partitions should be equal or multiple of the number of executors on Query applications
+    * @param force               if true, merges even if there's only one  subpartition maxOpIndex=N
+    */
+  def compactLandingPartition(partitionColumnName: String, partitionValue: String, numPartitions: Int, force: Boolean) = {
+    println(s"Merging partition $partitionColumnName=$partitionValue into $numPartitions files")
+
+    val lockFilePath = lockPath()
+
+    if (acquireLock(lockFilePath)) {
+      val startTs = new Date().getTime
+
+      val landingPartitionPath = s"${PathUtils.landingURI}/$partitionColumnName=$partitionValue"
+
+      try {
+        val subPartToCompact = getSubpartitionsToCompact(landingPartitionPath, force)
+
+        if (subPartToCompact.nonEmpty) {
+
+          var data = spark.read
+            .schema(CluesoConstants.storedEventSchema
+              .add("maxOpIndex", LongType)
+            )
+            .parquet(PathUtils.landingURI)
+            .where(col(partitionColumnName) === lit(partitionValue))
+            .where(col("maxOpIndex").isin(subPartToCompact: _*))
+
+          // window function over union of partitions bucketName=<specified bucketName>
+          val windowSpec = Window.partitionBy("bucket","key").orderBy(col("kafkaTimestamp").desc)
+
+          data.coalesce(numPartitions)
+            .withColumn("rank", dense_rank().over(windowSpec))
+            .where((col("rank") === 1).and(col("type") =!= "delete"))
+            .drop("rank")
+            .write
+            .partitionBy("bucket", "maxOpIndex")
+            .mode(SaveMode.Append)
+            .parquet(PathUtils.stagingURI)
+        }
+
```

Comment:
log that successfully compacted PathUtils.landingURI?

Reply:
(y)